File: clj-from-java/src/main/clojure/capclug/demo/cljfromjava/six.clj
(ns capclug.demo.cljfromjava.six
  (:gen-class
    :methods [[latida [Object Object] void]
              [datila [String int] void]]
              [foo [int String] void]))  ; this doesn't match the type hints!!???

(defn -latida [this #^String msg, #^Integer/TYPE n]
  (println (repeat n msg)))

(defn -datila [this msg n]
  (println (repeat n msg)))

(defn -foo [this #^String msg #^Integer/TYPE n]
  (println (repeat n msg)))

File: java-call-clj/src/main/java/capclug/demo/callclj/Six.java
/*
 * $Id$
 * Copyright (C) 2010 ARIN
 */
package capclug.demo.callclj;

import capclug.demo.cljfromjava.six;

/**
 * @version $Rev$
 */
public class Six
{
    public static void main( String[] args )
    {
        six six = new six();
        six.datila( "ooohh", 4 );
        // six.foo( "huh?", 2 ); // silently didn't compile????
        six.latida( 3, "uggh" ); // runtime exception!
    }
}
